### ISSUE TITLE
3.0 - Audit filters in Structured_Data class

### ISSUE THREAD

**Issue by ashleyfae** [easydigitaldownloads/easy-digital-downloads#8818]
## Bug Report

The `EDD\Structured_Data` class is new in 3.0. There are _three_ filters here for modifying the structured data:

1. The `get_data()` method has `edd_structured_data_get_data`. This is a filter on the data when it's retrieved. Note that it's filtering an array of arrays. (By default there's only one item in the array --- EDD's data --- but the class is designed so that others can be added.
2. The `set_data()` method has `edd_structured_data_set_data`, which is a filter on the data as it's being set.
3. The `generate_download_data()` method has `edd_generate_download_structured_data`, which runs the data through a filter _just_ prior to calling `set_data()`.

There's also an action hook `edd_generate_structured_data`, which seems to be intended for allowing additional data to be added, which is fine, except I also feel like that could be equally achieved via the filters.

tl;dr: There's a lot going on here for accomplishing _almost_ the same thing. I think I understand the intended differences:

1. `edd_structured_data_get_data` is a filter on the array of arrays. (Though I feel like this might be very similar to the action hook's use case?)
2. `edd_structured_data_set_data` is a filter on one array that's being set, but you don't know for sure that it's EDD's array.
3. `edd_generate_download_structured_data` is a filter on EDD's structured data that's being set.

But IMO these filters are accounting for use cases that don't really exist yet. For now, I think it actually leads to some confusion. In fact, I opened this issue because I was going to do a PR in EDD Reviews for modifying the data and I was confused about which filter to use.

I think for now we keep the action hook and the filter on `edd_generate_download_structured_data` and remove the others. We can always add more filters if we get requests for them, but I'd rather not add too many preemptively because it can make code harder to refactor later down the line.

### Information (if a specific version is affected):
PHP Version:

EDD Version (or branch): release/3.0

WordPress Version:

Any other relevant information:
